Changes

```diff
@@ -1,6 +1,59 @@
 OCaml 4.07
 ----------
 
+Highlights of this release include:
+
+- The way the standard library modules are organized internally has
+  changed (GPR#1010):
+  1. the List module (for example) is now named Stdlib__list
+  2. a new Stdlib module contains a series of aliases
+  such as (module List = Stdlib__list)
+  3. the Stdlib module is implicitly opened when type-checking OCaml
+  programs (as Pervasives previously was), so that Stdlib.List can be
+  accessed as just List, as before.
+
+  This should be invisible to most users, although it is possible that
+  some tools show the Stdlib. or Stdlib__ prefixes in messages. (You
+  might want to report these situations as usability bugs.) The change
+  prevents standard library modules from conflicting with end-user
+  filenames (please avoid stdlib.ml and the stdlib__ prefix); we may
+  introduce new standard library modules in the future with less fears
+  of breaking user code. In particular, Float (GPR#1638) and Seq
+  (GPR#1002) modules have already been added to the standard library.
+
+- The error messages caused by various typing errors have been
+  improved to be easier to understand, in particular for
+  beginners. (GPR#1505, GPR#1510)
+
+  For example,
+  >  # while 1 do () done;;
+  >          ^
+  >  Error: This expression has type int but an expression was expected of type
+  >         bool
+  now adds the extra explanation
+  >  because it is in the condition of a while-loop
+
+- Effort has been made to reduce the compilation time of flambda
+  programs, and the size of the produced `.cmx` files when using
+  the -Oclassic optimisation level. (GPR#1401, GPR#1455, GPR#1627,
+  GPR#1665, GPR#1667)
+
+- The HTML manual has benefited from various style improvements;
+  it should look visually nicer than previous editions.
+  (GPR#1741, GPR#1757, GPR#1767)
+
+- Since 4.01, it is possible to select a variant constructor or
+  record field from a sub-module that is not opened in the current
+  scope, if type information is available at the point of use. This
+  now also works for GADT constructors. (GPR#1648)
+
+- The GC should handle better the accumulation of custom blocks in the minor
+  heap; this should solve some memory-usage issues observed by code
+  which allocates a lot of small custom blocks (small bigarrays)
```

Comment:
The GC should handle ~better~ the ... minor heap **better**; this should ... observed ~by~ in code which .. blocks (**e.g.** small bigarrays)?

Reply:
Thanks! `e.g.` is jargon, I try to avoid it, but I reformulated.

Reply:
Come on, you're not a proper academic until you use ***e.g.*** and ***i.e.*** every other sentence.  Extra points for using ***viz.*** and ***ibid.*** as well :-)

Reply:
I'll show you:

OCaml, i.e. the functional programming language of choice, offers full support for typed programming goodness, e.g. GADTs and polymorphism (Damas and Milner, 1982), viz. the ability to use a definition in several, differently-typed contexts (ibid.).

Reply:
I refuse the accept Latin as being jargon :wink:

Reply:
Joking aside, your re-wording is nicer - I just prefer to have “for example,”, “e.g.” or, in extremis, “exempli gratia,” if one is giving a non-exhaustive list of examples!